server/bin/create_test_repos.py

```diff
@@ -0,0 +1,469 @@
+
+    # Print output of command
+    if verbose is True:
+        for line in process.stdout.readlines():
+            print line
+
+    # Wait for result of process
+    ret = process.wait()
+
+    return ret
+
+
+def read_test_data(filename):
+    """
+    Try to read test data from json file.
+    """
+    try:
+        with open(filename) as fp:
+            try:
+                data = json.load(fp)
+            except ValueError as err:
+                print("Unable to read file %s with test data: %s" % (filename, str(err)))
+                return None
+    except IOError as err:
+        print("Unable to read file: %s with test data: %s" % (filename, str(err)))
+        return None
+    return data
+
+
+def get_repo_definitions(test_data):
+    """
+    Get list of repositories from test_data
+    """
+
+    if test_data is None:
+        return []
+
+    try:
+        repo_definitions = test_data['content']
+    except KeyError as err:
+        print('Info: Test data does not include any global definition of repositories')
+        repo_definitions = []
+
+    # There can be also some content specific for owners
+    try:
+        owners = test_data['owners']
+    except KeyError as err:
+        return repo_definitions
+    
+    for owner in owners:
+        try:
+            owner_content = owner['content']
+        except KeyError as err:
+            continue
+        else:
+            repo_definitions.extend(owner_content)
+    
+    return repo_definitions
+
+
+def add_packages_to_repo(packages_list, repo_path, repository):
+    """
+    This function tries to add list of packages to the repository
+    """
+    for pkg_name in packages_list:
+        print("\tAdding package %s" % pkg_name)
+        pkg_file_name = pkg_name + '-1-0.noarch.rpm'
+        src_pkg_file_path = RPMBUILD_ROOT_DIR + '/RPMS/noarch/' + pkg_file_name
+        dst_pkg_file_path = repo_path + '/RPMS/' + pkg_file_name
+        shutil.copyfile(src_pkg_file_path, dst_pkg_file_path)
+
+
+def generate_repositories(repo_definitions):
+    """
+    This function tries to generate yum repositories with some dummy packages
+    """
+    if repo_definitions is None:
+        return None
+
+    # Make sure that root directory of repositories exist
+    try:
+        if not os.path.exists(REPO_ROOT_DIR):
+            os.makedirs(REPO_ROOT_DIR)
+    except OSError as err:
+        print("Unable to create directory: %s, %s" % (REPO_ROOT_DIR, str(err)))
+        return None
+
+    for repo in repo_definitions:
+        repo_path = REPO_ROOT_DIR + repo['content_url']
+        print("creating repository %s in %s" % (repo['name'], repo_path ))
+        if not os.path.exists(repo_path + '/RPMS'):
+            os.makedirs(repo_path + '/RPMS')
+
+        add_packages_to_repo(repo['packages'], repo_path, repo)
+
+        # Create repository with RPM packages
+        run_command('createrepo %s' % repo_path)
```

Comment:
This is just a feeling I have, but I get the sense that `createrepo_c` is the way the wind is blowing.  It's under active development and it doesn't require yum as `createrepo` does.

Reply:
Interesting. The `createrepo` is just symlink on `createrepo_c` on Fedora 29. The `createrepo` is bash script from different rpm package on CentOS 7 (we use it in our Vagrant environment). Syntax should be almost the same, but `createrepo` command is probably considered deprecated. Good point. Thanks.